File: Labs/C++/Lab_13/Problem_01.cpp
// Lab 13 - Problem 1
// Derek P Sifford
#include <iostream>
using namespace std;

void printArray(int myArr[], int size) {
    for (int i = 0; i < size; i++) {
        cout << myArr[i] << endl;
    }
}

int smallestIndex(int myArr[], int size) {
    int output = 0;
    for (int i = 0; i < size; i++) {
        if (myArr[i] < output) {
            output = myArr[i];
        }
    }
    return output;
}

int main() {

    int myArr[5] = {0,1,2,3,4};
    int smallest;

    printArray(myArr, 5);
    smallest = smallestIndex(myArr, 5);

    cout << "\nThe smallest index of the array is " << smallest << endl;

    return 0;
}

File: Labs/golang/Lab_13/Problem_01.go
package main

import (
	"fmt"
)

// Problem1 - Write two functions:
//     1) printArray: Prints an int array
//     2) smallestIndex: takes as parameters an int array and its size and
//        returns the index of the first occurrence of the smallest element
//        in the array.
func Problem1() {
	fmt.Printf("Problem 1:\n")
	mySlice := []int{0, 1, 2, 3, 4, -10}
	printArray(mySlice)
	smallest := smallestIndex(mySlice)
	fmt.Printf("Smallest index = %v\n", smallest)
}

func printArray(s []int) {
	fmt.Printf("%+v\n", s)
}

func smallestIndex(s []int) (smallest int) {
	for _, i := range s {
		if i < smallest {
			smallest = i
		}
	}
	return
}
